In [1]:
import time
import numpy as np
from scipy.interpolate import griddata

def func(x, y):
    return x*(1-x)*np.cos(4*np.pi*x) * np.sin(4*np.pi*y**2)**2

grid_x, grid_y = np.mgrid[0:1:20000j, 0:1:20000j]

points = np.random.rand(1000, 2)
values = func(points[:,0], points[:,1])

tic = time.perf_counter()
grid_z = griddata(points, values, (grid_x, grid_y), method='linear')
toc = time.perf_counter()

print(f"Time to compute griddata interpolation: {toc - tic:0.4f} seconds")

Time to compute griddata interpolation: 21.9338 seconds


gs674-ddezeeuw
Time to compute griddata interpolation: 21.9338 seconds

cloud_dev7
Time to compute griddata interpolation: 56.5639 seconds

kahala2
Time to compute griddata interpolation: 44.9300 seconds

webdev
--memory error
